src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -39,6 +39,29 @@ public string GetString()
             return JsonReaderHelper.TranscodeHelper(span);
         }
 
+        /// <summary>
+        /// Reads value of the next JSON token for a comment from the source transcoded as a <see cref="string"/>,
```

Comment:
Based on https://github.com/dotnet/corefx/pull/35705#discussion_r261847380, let's not preserve the comment delimiters.

Reply:
Would it make sense to strip the comment delimiters (`\\`, `/*` and `*/`) and the line separators (`\r`, `\r\n` and `\n`) during parsing where the `ValueSpan` or `ValueSequence` for the `JsonTokenType.Comment` is prepared? In that case, should I do the changes in PR #33393 or should I do them in this PR? If it is to be done in this PR, then it would help if we have an early closure on #33393 so that this code can be rebased to include work for `\r\n`.

Reply:
> Would it make sense to strip the comment delimiters (`\\`, `/*` and `*/`) and the line separators (`\r`, `\r\n` and `\n`) during parsing where the `ValueSpan` or `ValueSequence` for the `JsonTokenType.Comment` is prepared? 

Yes, it would.

> In that case, should I do the changes in PR #33393 or should I do them in this PR?

Let's stage it in smaller, clearer chunks, where the change is isolated. So, in this order:
1) https://github.com/dotnet/corefx/pull/33393 - Add support for single line comments ending on \r, \r\n
2) Open up a new PR - Change ValueSpan/ValueSequence to exclude the comment delimiters, effectively undoing **parts of** https://github.com/dotnet/corefx/pull/33216/commits/4a2aa632e3e724f64c9d47274e0cb2e5a96b2ff8
3) This PR (https://github.com/dotnet/corefx/pull/35705) to expose new GetComment API

What do you think of this approach?

> Query: IIUC this [line](https://github.com/dotnet/corefx/blob/9074d9f5b513a95b7eac9ba1112069c29230d1d2/src/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs#L20) in `JsonReaderHelper.CountNewLines` needs to handle `\r` on its own as a line separator, and also ensure that `\r\n` are handled properly... ?

Unless this is blocking some of the work above, I would file an issue for this and investigate/fix separately. Some questions we likely want to resolve first: What does Json.NET return for line number when comments end with just `\r` (for Windows vs Unix)? Do we double count line number `\r\n` or treat it as a single line increment?

Reply:
> Let's stage it in smaller, clearer chunks, where the change is isolated. So, in this order:
> 
> 1. #33393 - Add support for single line comments ending on \r, \r\n
> 2. Open up a new PR - Change ValueSpan/ValueSequence to exclude the comment delimiters, effectively undoing **parts of** [4a2aa63](https://github.com/dotnet/corefx/commit/4a2aa632e3e724f64c9d47274e0cb2e5a96b2ff8) from #33216
> 3. This PR (#35705) to expose new GetComment API
> 
> What do you think of this approach?

Sounds like a plan! Thanks! So, I am working on #33393 on priority to finish step 1, and then pull in the code changes, once submitted into master, into step 2. So will put this (and new) PR a bit on hold. Kindly see updates for #33393.

Reply:
> Do we double count line number \r\n or treat it as a single line increment?

Json.NET treats it as a single line increment